src/main/fc/fc_msp.c

```diff
@@ -2038,31 +2042,53 @@ static mspResult_e mspCommonProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
 #if defined(OSD) || defined (USE_OSD_SLAVE)
     case MSP_SET_OSD_CONFIG:
         {
-            const uint8_t addr = sbufReadU8(src);
-            // set all the other settings
-            if ((int8_t)addr == -1) {
+            const uint8_t addr1 = sbufReadU8(src);
+            const uint8_t addr2 = sbufReadU8(src);
```

Comment:
If that is the plan, this should still be possible while retaining compatibility to the old API: Add in an index for the screen layout (with a reserved value for the stats screen) after the uint16 for the position. If the client doesn't send it, you know it's a legacy client, configuring the first screen.

(One assumption in the design of the MSP messages is that the receiver reads the values it knows about from the message, and ignores any trailing data. This way it is possible to extend the protocol by adding values at the end, without breaking clients that were released before the change.)